warehouse/core/src/main/java/datawave/ingest/mapreduce/job/SplittableRFileRecordReader.java

```diff
@@ -0,0 +1,99 @@
+package datawave.ingest.mapreduce.job;
+
+import static org.apache.accumulo.core.conf.Property.TABLE_CRYPTO_PREFIX;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.util.Collections;
+
+import org.apache.accumulo.core.crypto.CryptoFactoryLoader;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.file.FileSKVIterator;
+import org.apache.accumulo.core.file.blockfile.impl.CachableBlockFile;
+import org.apache.accumulo.core.file.rfile.RFile;
+import org.apache.accumulo.core.spi.crypto.CryptoEnvironment;
+import org.apache.accumulo.core.spi.crypto.CryptoService;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+public class SplittableRFileRecordReader extends RFileRecordReader {
+    @Override
+    public void initialize(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+        if (split instanceof RFileSplit) {
```

Comment:
Consider commenting on the cases when an `RFileSplit` is expected versus some other type of `InputSplit`, it was a little difficult to trace through the code and figure this out.

Also, `RFileRecordReader` makes an assumption in its `initialize` method that `fileSplit.getStart()` will always return 0 and throws an `IOException` if it doesn't. I don't think that will cause problems in this case, but not 100% certain.

Reply:
I did notice that because I'm calling super.getSplits(job) to get the file splits and I've overridden isSplittable() to return true there actually are pre-split rfiles returned. These will need to be de-duped/discarded in the SplittableRFileInputFormat. Good catch.